File: warmupw6d1/switchup_warmup.js

// # Switch up
// # Write a function switchUp that takes a string as an argument and returns the string with the case for each letter switched e.g. lowercase letter must become uppercase and vice versa.

// # switchUp("Switch caSe FoR eVery letter") --> // "sWITCH CAsE fOr EvERY LETTER"
// # Extension
// # If words are separated by spaces or underscores, change them to dashes. For example:

// # switchUp2("js_Was_created_bY BRENdan Eich") --> // "JS-wAS-CREATED-By-brenDAN-eICH"

    
var switchUp = str => {
    arr = [];
    str.split('').forEach(char => {
        switch(true) {
            case /[A-Z]/.test(char):
                arr.push(char.toLowerCase());
                break;
            case /[a-z]/.test(char):
                arr.push(char.toUpperCase());
                break;
            case /[\s,_]/.test(char):
                arr.push('-');
                break;
        }
   });
    newStr = arr.join('');
    return newStr; 
}

console.log(switchUp("js_Was_created_bY BRENdan Eich"));

File: warmupw6d1/switchup_warmup.rb
# Switch up
# Write a function switchUp that takes a string as an argument and returns the string with the case for each letter switched e.g. lowercase letter must become uppercase and vice versa.

# switchUp("Switch caSe FoR eVery letter") --> // "sWITCH CAsE fOr EvERY LETTER"
# Extension
# If words are separated by spaces or underscores, change them to dashes. For example:

# switchUp2("js_Was_created_bY BRENdan Eich") --> // "JS-wAS-CREATED-By-brenDAN-eICH"


require 'pry'

def switch_up(str)
    arr = []
    new_arr = str.split('').map do |char|
        case char 
        when /[A-Z]/
            char.downcase!
        when /[a-z]/
           char.upcase!
        when /[\s,_]/   
            "-"
        end
    end
    switched = new_arr.join()
    return switched
    
end

puts switch_up("js_Was_created_bY BRENdan Eich")